packages/material-ui-lab/src/SpeedDialAction/SpeedDialAction.js

```diff
@@ -36,14 +36,32 @@ class SpeedDialAction extends React.Component {
     tooltipOpen: false,
   };
 
+  static getDerivedStateFromProps = (props, state) => {
+    if (!props.open && state.tooltipOpen) {
+      state.tooltipOpen = false;
+    }
+    return state;
+  };
```

Comment:
That is not how [`getDerivedStateFromProps`](https://reactjs.org/docs/react-component.html#static-getderivedstatefromprops) works.

Return `null` for no change or an object that should be merged.